### PR TITLE
[BLOG] Add pytorch-tensoriterator-internals

### PR DESCRIPTION
**Pull request by MarsBarLee**
This blog post is being resubmitted with the new workflow:

- Branching directly from the Quansight-website repository instead of a fork
- which allows me to authorize deployment of Vercel to build preview URLs
- making it easier to catch errors and make changes as necessary

The previous blog post PR for pytorch-tensoriterator-internals, #362, is closed.

## Text styling

- [x] The blog is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags in order to do so (with only one level-one header).
- [x] All links describe where they link to (for example, check the [Quansight labs website](https://labs.quansight.org/)).
- [x] Any kind of styling that the author uses (for example, bold for emphasis) is consistent throughout the blog.

## Non-text contents

- [x] All content is represented as text (for example, images need alt text and videos need captions or descriptive transcripts).
- [x] If there are emojis, there are not more than three in a row.
- [x] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [x] If it were to be read as plain text, the blog still makes sense and no information is missing.

## Links to other Quansight blogs:
- in the blog post, the link is internally linked as `/blog/2021/04/pytorch-tensoriterator-internals-update/index.html` in the .alert-warning at the top of the post
- Keeping the internal link gave a build error
- I changed the link to `https://labs.quansight.org/blog/2021/04/pytorch-tensoriterator-internals-update`, even though `https://labs.quansight.org/blog/2021/04/pytorch-tensoriterator-internals-update` has not migrated

no llc links found

### CODE REVIEW

File: apps/labs/posts/pytorch-tensoriterator-internals.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+author: sameer-deshmukh
+description: 'This post is a deep dive into how TensorIterator works, and is an essential part of learning to contribute to the PyTorch codebase since iterations over tensors in the C++ codebase are extremely commonplace. This post is aimed at someone who wants to contribute to PyTorch'
+category: [Machine Learning]
+featuredImage:
+  src: posts/pytorch-tensoriterator-internals/feature.png
+  alt: 'Code snippet on creating a TensorIterator using a default constructor.'
+hero:
+  imageSrc: posts/pytorch-tensoriterator-internals/blog_hero_var2.svg
+  imageAlt: 'An illustration of a dark brown hand holding up a microphone, with some graphical elements highlighting the top of the microphone.'
+'
```
Comment (MarsBarLee) [inline apps/labs/posts/pytorch-tensoriterator-internals.md:13]:
```suggestion
```

Based on r[ecent Vercel build error](https://vercel.com/quansight/quansight-labs-live/2Z7sZW85bn2byHAAS2siN5nS4XCU), `error YAMLException: unexpected end of the stream within a single quoted scalar at line 14, column 1:`

File: apps/labs/posts/pytorch-tensoriterator-internals.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+hero:
+  imageSrc: posts/pytorch-tensoriterator-internals/blog_hero_var2.svg
+  imageAlt: 'An illustration of a dark brown hand holding up a microphone, with some graphical elements highlighting the top of the microphone.'
+---
+
+The history section of this post is still relevant, but `TensorIterator`'s
+interface has changed significantly. For an update on the new API, please check
+out [this new blog
+post](https://labs.quansight.org/blog/2021/04/pytorch-tensoriterator-internals-update).
+{: .alert .alert-warning}
```
Comment (MarsBarLee) [inline apps/labs/posts/pytorch-tensoriterator-internals.md:19]:
```suggestion
```

Based on [recent Vercel build error](https://vercel.com/quansight/quansight-labs-live/PyjRvapNYZsrz2tVtra2NSN7gKBe), `error [Error: [next-mdx-remote] error compiling MDX:`

File: apps/labs/posts/pytorch-tensoriterator-internals.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+title: 'PyTorch TensorIterator Internals'
+published: April 13, 2020
+author: sameer-deshmukh
+description: 'This post is a deep dive into how TensorIterator works, and is an essential part of learning to contribute to the PyTorch codebase since iterations over tensors in the C++ codebase are extremely commonplace. This post is aimed at someone who wants to contribute to PyTorch'
+category: [Machine Learning]
+featuredImage:
+  src: posts/pytorch-tensoriterator-internals/feature.png
+  alt: 'Code snippet on creating a TensorIterator using a default constructor.'
+hero:
+  imageSrc: posts/pytorch-tensoriterator-internals/blog_hero_var2.svg
```
Comment (MarsBarLee) [inline apps/labs/posts/pytorch-tensoriterator-internals.md:11]:
```suggestion
  imageSrc: posts/pytorch-tensoriterator-internals/blog_hero_var1.svg
```

Change file path to fix blank image, based on [recent blog post build](https://quansight-labs-live-git-pytorch-tensoriterator-3c59ee-quansight.vercel.app/blog/pytorch-tensoriterator-internals).

File: apps/labs/posts/pytorch-tensoriterator-internals.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+  alt: 'Code snippet on creating a TensorIterator using a default constructor.'
+hero:
+  imageSrc: posts/pytorch-tensoriterator-internals/blog_hero_var1.svg
+  imageAlt: 'An illustration of a dark brown hand holding up a microphone, with some graphical elements highlighting the top of the microphone.'
+---
+
+> The history section of this post is still relevant, but `TensorIterator`'s
+> interface has changed significantly. For an update on the new API, please check
+> out [this new blog
+> post](https://labs.quansight.org/blog/2021/04/pytorch-tensoriterator-internals-update).
```
Comment (MarsBarLee) [inline apps/labs/posts/pytorch-tensoriterator-internals.md:18]: Per my discussion with Gabriel, there is no equivalent to {: .alert .alert-warning} from the old website styling.

We decided on block quotes for now, as there is no built-in alert function or styling in the new website.

Possible long term solutions we disucssed: MDX plugin or custom tailwind styling

**Comment by MarsBarLee** [comment]
@gabalafou Ready for review! Hero image loads, solved build errors, changed alert message.

File: apps/labs/posts/pytorch-tensoriterator-internals.md (hunk 4 above)
Comment (gabalafou) [reply apps/labs/posts/pytorch-tensoriterator-internals.md:18]: For future reference, here's the link to the blog post on the old site, to see how this paragraph used to be styled:
https://quansight-labs.netlify.app/blog/2020/04/pytorch-tensoriterator-internals/

**Pull request merged**